lib/ReactViews/FeatureInfo/FeatureInfoPanel.jsx

```diff
@@ -119,6 +122,58 @@ const FeatureInfoPanel = createReactClass({
         }
     },
 
+    addManualMarker(latitude, longitude) {
+        addMarker(this.props.terria, {
+            name: "User Selection",
+            location: {
+                latitude: latitude,
+                longitude: longitude
+            }
+        });
+    },
+
+    pinClicked(latitude, longitude) {
+        if (!markerVisible(this.props.terria)) {
+            this.addManualMarker(latitude, longitude);
+        } else {
+            removeMarker(this.props.terria);
+        }
+    },
+
+    locationUpdated(latitude, longitude) {
+        if (defined(latitude) && defined (longitude) && markerVisible(this.props.terria)) {
+            removeMarker(this.props.terria);
+            this.addManualMarker(latitude, longitude);
+        }
+    },
+
+    locationItem(cartesianPosition) {
```

Comment:
How about `renderLocationItem`?  In general, functions should be named as verbs.